reframe/core/logging.py

```diff
@@ -378,6 +380,24 @@ def _update_check_extras(self):
         if self.check.job:
             self.extra['check_jobid'] = self.check.job.jobid
 
+        try:
+            import pwd
+            self.extra['username'] = pwd.getpwuid(os.geteuid()).pw_name
+        except:
+            try:
+                import getpass
+                self.extra['username'] = getpass.getuser()
+            except:
+                pass
+
+        try:
+            import grp
+            import pwd
+            gid = pwd.getpwnam(self.extra['username']).pw_gid
+            self.extra['group'] = grp.getgrgid(gid).gr_name
+        except:
+            pass
```

Comment:
The whole thing here is overly complicated. I suggest the following:

```python
try:
    user = getpass.getuser()
except Exception:
    user = '<unknown>'

try:
    group = grp.getgrgid(os.getgid()).gr_name
except KeyError:
    group = '<unknown>'

self.extra['osuser'] = user
self.extra['osgroup'] = group
```

The `getpass()` does already use `pwd` under the hood if necessary.

We would the extreme way proposed here only if we had actually to care about effective and real user ids. But (a) I don't think we have to and (b) if that was the case, this would have to go in `utility.os_ext`.

Reply:
Also @victorusu, I indeed think this should not go in the `_update_check_extras`. It should be in the `LoggerAdapter`, but the I suggest moving this query to the `os_ext.utility` as `osuser()` and `osgroup()` functions. These functions should return the user and group name or `None`, so that inside the `LoggerAdapter` you would do the following:

```python
                'osuser': os_ext.osuser() or '<unknown>',
                'osgroup': os_ext.osgroup() or '<unknown>',
```

Reply:
There are a couple of places in the framework that we are using `os.environ['USER']` to retrieve the current user. We should replace these with the new utility function. Sorry, you asked for it! 😛